.github/workflows/unit-tests.yml

```diff
@@ -65,11 +65,50 @@ jobs:
             echo EOF
           } >> "${GITHUB_OUTPUT}"
 
+  retrieve-selected-tests:
```

Comment:
The logic looks good.

I am wondering whether moving this to a step in the `retrieve-commit-message` would be a good idea.
- upside: all the commit related info is in the same job. reusing the same job rather than launching a new one so a tiny bit quicker (but honestly not that crucial)
- downside: you probably need to export the commit message as environment variable into both `GITHUB_ENV` and `GITHUB_OUTPUT` but that seems fine.